langchain/README.md

```diff
@@ -1,53 +1,243 @@
+
+   ```shell
+   pip install ddtrace
+   ```
+
+2. Start your application with the following command, enabling agentless mode:
+
+   ```shell
+    DD_SITE=<YOUR_DATADOG_SITE> DD_API_KEY=<YOUR_API_KEY> DD_LLMOBS_ENABLED=1 DD_LLMOBS_AGENTLESS_ENABLED=1 DD_LLMOBS_ML_APP=<YOUR_ML_APP_NAME> ddtrace-run python <your_app>.py
+   ```
+
+##### If you already have the Datadog Agent installed:
+1. Make sure the Agent is running and that APM and StatsD are enabled. For example, use the following command with Docker:
+
+   ```shell
+   docker run -d \
+     --cgroupns host \
+     --pid host \
+     -v /var/run/docker.sock:/var/run/docker.sock:ro \
+     -v /proc/:/host/proc/:ro \
+     -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
+     -e DD_API_KEY=<DATADOG_API_KEY> \
+     -p 127.0.0.1:8126:8126/tcp \
+     -p 127.0.0.1:8125:8125/udp \
+     -e DD_DOGSTATSD_NON_LOCAL_TRAFFIC=true \
+     -e DD_APM_ENABLED=true \
+     gcr.io/datadoghq/agent:latest
+   ```
+
+2. Install the `ddtrace` package if it isn't installed yet:
+
+   ```shell
+   pip install ddtrace
+   ```
+
+3. Start your application using the `ddtrace-run` command to automatically enable tracing:
+
+   ```shell
+     DD_SITE=<YOUR_DATADOG_SITE> DD_API_KEY=<YOUR_API_KEY> DD_LLMOBS_ENABLED=1 DD_LLMOBS_ML_APP=<YOUR_ML_APP_NAME> ddtrace-run python <your_app>.py
+   ```
+
+**Note**: If the Agent is running on a custom host or port, set `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT` accordingly.
+
+##### If you are running LLM Observability in a serverless environment (AWS Lambda):
+1. Install the **Datadog-Python** and **Datadog-Extension** Lambda layers as part of your AWS Lambda setup.
+2. Enable LLM Observability by setting the following environment variables:
+
+   ```shell
+    DD_SITE=<YOUR_DATADOG_SITE> DD_API_KEY=<YOUR_API_KEY> DD_LLMOBS_ENABLED=1 DD_LLMOBS_ML_APP=<YOUR_ML_APP_NAME>
+   ```
+
+**Note**: In serverless environments, Datadog automatically flushes spans at the end of the Lambda function.
+
+##### Automatic LangChain Tracing
+
+LangChain integration is automatically enabled when LLM Observability is configured. This captures latency, errors, input/output messages, and token usage for LangChain operations.
+
+The following methods are traced for both synchronous and asynchronous LangChain operations:
+
+- LLMs: `llm.invoke()`, `llm.ainvoke()`
+- Chat Models: `chat_model.invoke()`, `chat_model.ainvoke()`
+- Chains/LCEL: `chain.invoke()`, `chain.ainvoke()`, `chain.batch()`, `chain.abatch()`
+- Embeddings: `OpenAIEmbeddings.embed_documents()`, `OpenAIEmbeddings.embed_query()`
+
+No additional setup is required for these methods.
+
+##### Validation
 
+Validate that LLM Observability is properly capturing spans by checking your application logs for successful span creation. You can also run the following command to check the status of the `ddtrace` integration:
+
+```shell
+ddtrace-run --info
+```
+
+Look for the following message to confirm the setup:
+
+```
+Agent error: None
 ```
```

Comment:
```suggestion
```
These steps are identical to the validation steps below. Let's keep them at the bottom.